### PR TITLE
[BUG] Regression base class score function

### LINKED ISSUE #533: [BUG] BaseRegressor bug when used with score
### Describe the bug

BaseRegressor crashes when using score. The scikit method r2_score has no attribute normalize, I'm guessing it was deprecated some time ago. This is legacy code not tested for in the base class. The score function

```python

    def score(self, X, y) -> float:
        """Scores predicted labels against ground truth labels on X.

        Parameters
        ----------
        X : 3D np.array (any number of channels, equal length series)
                of shape [n_instances, n_channels, series_length]
            or 2D np.array (univariate, equal length series)
                of shape [n_instances, series_length]
            or of any other supported type
                for list of mtypes, see datatypes.SCITYPE_REGISTER
        y : 1D np.array of float, of shape [n_instances] - regression labels (gnr truth)
            indices correspond to instance indices in X

        Returns
        -------
        float, R-squared score of predict(X) vs y
        """
        from sklearn.metrics import r2_score

        self.check_is_fitted()

        return r2_score(y, self.predict(X), normalize=True)
```

scikit method has this signature

```python
def r2_score(
    y_true,
    y_pred,
    *,
    sample_weight=None,
    multioutput="uniform_average",
    force_finite=True,
):
```
maybe normalize is from a different package such as statsmodels? 

### Steps/Code to reproduce the bug

```python
from aeon.regression.distance_based import KNeighborsTimeSeriesRegressor
from aeon.datasets import load_covid_3month
covid_train, covid_trainy = load_covid_3month(split="train")
covid_test, covid_testy = load_covid_3month(split="test")
knn = KNeighborsTimeSeriesRegressor()
knn.fit(covid_train, covid_trainy)
knn.score(covid_test,covid_testy)
```

### Expected results

Correct scoring

### Actual results

```python-trackback
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[13], line 7
      5 knn = KNeighborsTimeSeriesRegressor()
      6 knn.fit(covid_train, covid_trainy)
----> 7 knn.score(covid_test,covid_testy)

File C:\Code\aeon\aeon\regression\base.py:211, in BaseRegressor.score(self, X, y)
    207 from sklearn.metrics import r2_score
    209 self.check_is_fitted()
--> 211 return r2_score(y, self.predict(X), normalize=True)

TypeError: r2_score() got an unexpected keyword argument 'normalize
```

### Versions

_No response_

### PR DESCRIPTION
**Pull request by TonyBagnall**
fixes #533 

I have also moved the function _internal_convert up to BaseEstimator, since identical functions are in classifier and regressor, and functionality shared with clustering (to do later)

I also added a simple test for base regressor and reformatted some doc strings.

bit scope creep, moved DummyRegressor to match DummyClassifier

### CODE REVIEW

**Review by MatthewMiddlehurst — CHANGES_REQUESTED** [review]
I feel like there should be another class between BaseEstimator and the 3 ML base classes, but not going to push that here.

File: aeon/regression/base.py [hunk 1]
```diff
@@ -22,7 +22,6 @@ class name: BaseRegressor
 __all__ = [
     "BaseRegressor",
 ]
-__author__ = ["mloning", "fkiraly"]
```
[the remark this answers is not in the document]
Comment (TonyBagnall) [reply aeon/regression/base.py:25]: didnt realised I had removed it! I think I am in favour of having authors, but can get excessive

File: aeon/regression/base.py [hunk 2]
```diff
@@ -207,8 +213,11 @@ def score(self, X, y) -> float:
         from sklearn.metrics import r2_score
 
         self.check_is_fitted()
+        if isinstance(y, pd.Series):
+            y = pd.Series.to_numpy(y)
+        y = y.astype("float")
```
[the remark this answers is not in the document]
Comment (TonyBagnall) [reply aeon/regression/base.py:218]: I moved the method up to base, but the base classifier convert does not insist on float, which regression does, so it is enhancing the base estimator method. Could override it I guess

[the remark this answers is not in the document]
Comment (TonyBagnall) [reply aeon/regression/base.py:218]: ah sorry, thats in score, yes

[the remark this answers is not in the document]
Comment (TonyBagnall) [reply aeon/regression/base.py:218]: at least it actually works and is tested now :)

**Comment by TonyBagnall** [comment]
> I feel like there should be another class between BaseEstimator and the 3 ML base classes, but not going to push that here.

possibly bring this in with the unified validation, although dont know what to call it.

**Review by MatthewMiddlehurst — APPROVED** [review]

**Pull request merged**